### PR TITLE
Add DSC Configuration to automate environment setup 

### PR DESCRIPTION
**Pull request by acangialosi**
Added a DSC configuration yaml and associated instructions to use DSC to setup a complete working dev environment. Yaml was added to a .configurations folder so that it can also be automatically detected by Dev Home and later Dev Box for automatic environment setup.

### CODE REVIEW

**Comment by timheuer** [comment]
@adityamandaleeka FYI this adds a DevBox customization configuration to the eShop world...enabling eShop to be used as part of an included devbox image and have all the config needed to get it working on a new image.  No eShop app code touched in this, think of this like a Codespaces config added value.

**Comment by acangialosi** [comment]
@timheuer  & @adityamandaleeka. The configurations I added above will actually work on any windows development environment. They are only the DSC resources and powershell scripts. DSC is built into Windows.

File: .configurations/configuration.dsc.yaml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+       ensure: Present
+   ### Install VS Code Extension
+   ### -------------------------------------
+   - resource: PSDscResources/Script
+     id: vscodeextensions
+     dependsOn:
+       - vscode
+       - docker
+     directives:
+       description: Install Python Extension for VS Code
```
Comment (adityamandaleeka) [inline .configurations/configuration.dsc.yaml:117]: Python?

File: .configurations/configuration.dsc.yaml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+     settings:
+       SetScript: |
+         $env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine") + ";" + [System.Environment]::GetEnvironmentVariable("Path","User") 
+         code --install-extension github.copilot
```
Comment (adityamandaleeka) [inline .configurations/configuration.dsc.yaml:121]: Can we please scope this down to only the ones required to build/run eShop?

File: .configurations/configuration.dsc.yaml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+       productId: Microsoft.VisualStudio.Product.Community
+       channelId: VisualStudio.17.Preview
+       components: [Microsoft.VisualStudio.Workload.NetWeb, Microsoft.VisualStudio.Workload.NetCrossPlat, aspire]
+       includeRecommended: true
+   ### Install Microsoft Visual Studio Code
+   ### -------------------------------------
+   - resource: Microsoft.WinGet.DSC/WinGetPackage
+     id: vscode
+     directives:
+       description: Install Microsoft Visual Studio Code
```
Comment (adityamandaleeka) [inline .configurations/configuration.dsc.yaml:104]: Why does it install both VSCode and VS?

**Comment by adityamandaleeka** [comment]
Thanks for submitting this. I can see how this would be useful for some folks, but I feel it's a bit too aggressive in its current form and needs to be scoped down to just the minimum amount of things needed to work with eShop.

I'd also prefer to have the instructions show the manual steps first and then a "or, if you'd prefer, you can do this to have it set up for you". As a dev approaching the repo, it's useful to see what the setup process entails first, and this is especially true given that this is a sample and has educational value.

File: .configurations/configuration.dsc.yaml (hunk 1 above)
Comment (acangialosi) [reply .configurations/configuration.dsc.yaml:117]: That description was an inaccurate description that was copied over from another configuration. Python is not installed. Description is updated.

File: .configurations/configuration.dsc.yaml (hunk 2 above)
Comment (acangialosi) [reply .configurations/configuration.dsc.yaml:121]: I picked extensions that matched the tools in the readme. Copilot was the only tool not metnioned so I'll remove that one.

File: .configurations/configuration.dsc.yaml (hunk 3 above)
Comment (acangialosi) [reply .configurations/configuration.dsc.yaml:104]: I split this into a VSCode and a VSIDE configuration.

**Comment by acangialosi** [comment]
@adityamandaleeka , thanks for all the suggestions. I believe I resolved all your feedback.

**Comment by adityamandaleeka** [comment]
@acangialosi hmm, I don't see the actual updates. Did you forgot to push perhaps?

**Comment by acangialosi** [comment]
> @acangialosi hmm, I don't see the actual updates. Did you forget to push perhaps?

Yes, apologies. I merged my changes from my working branch back into the PR.

File: README.md [hunk 4]
```diff
@@ -20,6 +20,30 @@ A reference .NET application implementing an e-commerce website using a services
...
+```powershell
+install-Module -Name Microsoft.WinGet.Configuration -AllowPrerelease -AcceptLicense
+$env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine") + ";" + [System.Environment]::GetEnvironmentVariable("Path","User")
+get-WinGetConfiguration -file .\.configurations\vscode.dsc.yaml | Invoke-WinGetConfiguration -AcceptConfigurationAgreements
+shutdown /g
+```
+
+Or
+
+- From Dev Home go to `Machine Configuration -> Clone repositories`. Enter the URL for this reository. In the confirmation screen look for the section `Configuration File Detected` and click `Run File`.
```
Comment (adityamandaleeka) [inline README.md:45]:
```suggestion
- From Dev Home go to `Machine Configuration -> Clone repositories`. Enter the URL for this repository. In the confirmation screen look for the section `Configuration File Detected` and click `Run File`.
```

File: README.md [hunk 5]
```diff
@@ -20,6 +20,30 @@ A reference .NET application implementing an e-commerce website using a services
     - `.NET Aspire SDK` component in `Individual components`.
     - Optional: `.NET Multi-platform App UI development` to run client apps
 
+Or
+
+- Run the following commands in a Powershell & Terminal running as `Administrator` to automatically configuration your enviroment with the required tools to build and run this application. (Note: A restart is required and included in the script below.)
```
Comment (adityamandaleeka) [inline README.md:25]: > Run the following commands

Only one of the commands is needed right?

Comment (adityamandaleeka) [inline README.md:25]:
```suggestion
- Run the following commands in a Powershell & Terminal running as `Administrator` to automatically configure your enviroment with the required tools to build and run this application. (Note: A restart is required and included in the script below.)
```

File: README.md [hunk 6]
```diff
@@ -29,6 +43,17 @@ dotnet workload install aspire
 dotnet restore eShop.Web.slnf
 ```
 
+Or
+
+- Run the following commands in a Powershell & Terminal running as `Administrator` to automatically configuration your environment with the required tools to build and run this application. (Note: A restart is required after running the script below.)
+
+##### Install Visual Studio Code and related extensions
+```powershell
+install-Module -Name Microsoft.WinGet.Configuration -AllowPrerelease -AcceptLicense  -Force
```
Comment (adityamandaleeka) [inline README.md:52]: Why is the -Force required here?

Comment (acangialosi) [reply README.md:52]: It avoid an interrupting prompt for acceptance that otherwise blocks until the user confirms in the CLI.

**Comment by adityamandaleeka** [comment]
Sorry for the long delays @acangialosi... it's been a really busy few weeks. This seems reasonable to me. I'm a little bit concerned about the instructions telling people to run things as Admin and with the `-Force` option, but I will leave that to @blowdart to comment on based on other places where we provide such instructions.

**Review by adityamandaleeka — APPROVED** [review]

**Comment by adityamandaleeka** [comment]
Approved tentatively, want to give @blowdart a chance to comment on the above.

**Comment by blowdart** [comment]
Given that powershell can install extensions on a per user basis I'm confused about why admin privileges would be needed here. If it's just to reduce the number of elevation prompts during the winget restore then I'd remove the admin requirement and just let folks respond to each UAC prompt.

**Comment by acangialosi** [comment]
> Given that powershell can install extensions on a per user basis I'm confused about why admin privileges would be needed here. If it's just to reduce the number of elevation prompts during the winget restore then I'd remove the admin requirement and just let folks respond to each UAC prompt.

The PSDscResources/WindowsOptionalFeature DSC resources need to run as admin and Winget won't UAC prompt so the script just fails.

**Pull request merged**